### DISCUSSION TITLE
How to obtain the project list through API on version 3.0.0 rc2

### DISCUSSION THREAD

**Discussion by cristian-ciobanu** [GNS3/gns3-server#2447] [General]
I'm trying to get to list on projects from a GNS3 server 3.0.0 rc2 using curl and using default admin/admin credentials but always I'm receiving not authenticated message

I'm using the following command locally

`curl -X 'GET' -v -u admin:admin http://127.0.0.1:3080/v3/projects`

and the output is

```
*   Trying 127.0.0.1:3080...
* Connected to 127.0.0.1 (127.0.0.1) port 3080
* using HTTP/1.x
* Server auth using Basic with user 'admin'
GET /v3/projects HTTP/1.1
Host: 127.0.0.1:3080
Authorization: Basic YWRtaW46YWRtaW4=
User-Agent: curl/8.11.0
Accept: */*
 
* Request completely sent off
HTTP/1.1 401 Unauthorized
date: Tue, 26 Nov 2024 22:54:53 GMT
server: uvicorn
www-authenticate: Bearer
content-length: 31
content-type: application/json

* Connection #0 to host 127.0.0.1 left intact
{"message":"Not authenticated"}%
```
I am doing something wrong when using curl to query the API ?

**Comment by grossmj** [comment]
Hi Cristian,

We no longer use the HTTP Basic Authentication for the controller in version 3. We use JWT Token Based Authentication instead.

The API documentation is available at https://apiv3.gns3.net/ or locally when running the server: http://localhost:3080/docs

There are 2 endpoints available to authenticate an user: `/v3/access/users/login` or `/v3/access/users/authenticate`. They will return a token if the authentication is successful, you can then use that token for other API endpoints that require to be authenticated.

The best with CURL is to use the `/v3/access/users/authenticate` endpoint.

```
curl -X POST http://172.16.252.131/v3/access/users/authenticate -d '{"username": "admin", "password": "admin"}' -H "Content-Type: application/json"
{"access_token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhZG1pbiIsImV4cCI6MTczMjc1Nzg4OH0.WvYwMQM_4FQ1TUyh21n1zUadk5e-aW4dI3Kpy9-9ajA","token_type":"bearer"}
```

In this example, the token is `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhZG1pbiIsImV4cCI6MTczMjc1Nzg4OH0.WvYwMQM_4FQ1TUyh21n1zUadk5e-aW4dI3Kpy9-9ajA` and I think by default is valid for 1 day.

You need to include the token in the HTTP header like this: `curl http://172.16.252.131/v3/projects -H "Authorization: Bearer ${TOKEN}"`

```
curl http://172.16.252.131/v3/projects -H 'Accept: application/json' -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhZG1pbiIsImV4cCI6MTczMjc1Nzg4OH0.WvYwMQM_4FQ1TUyh21n1zUadk5e-aW4dI3Kpy9-9ajA"
[{"name":"MY_PROJECT","project_id":"b060e189-365b-4643-bb40-41c166720243","path":"/opt/gns3/projects/b060e189-365b-4643-bb40-41c166720243","auto_close":true,"auto_open":false,"auto_start":false,"scene_height":1000,"scene_width":2000,"zoom":100,"show_layers":false,"snap_to_grid":false,"show_grid":false,"grid_size":75,"drawing_grid_size":25,"show_interface_labels":false,"supplier":null,"variables":null,"status":"opened","filename":"MY_PROJECT.gns3"}]
```

**Comment by cristian-ciobanu** [comment]
Thank you. 
It works fine now with the steps suggested by you.
